File: desktop/src/types/importing.ts
export type MediaSource = {
  id: number;
  event_id: number;
  source_path: string;
  source_type: string;
  status: string;
  file_count: number;
  scanned_at: string | null;
  imported_at: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
};

export type ScanSourceSummary = {
  source_id: number;
  source_path: string;
  supported_files: number;
  unsupported_files: number;
  failed_files: number;
};

export type ScanResponse = {
  job_id: number;
  total_sources: number;
  supported_files: number;
  unsupported_files: number;
  failed_files: number;
  sources: ScanSourceSummary[];
};

export type ImportResponse = {
  job_id: number;
  imported_files: number;
  skipped_files: number;
  failed_files: number;
  total_files: number;
};

export type MetadataProcessResponse = {
  metadata_job_id: number;
  thumbnail_job_id: number;
  total_files: number;
  metadata_updated: number;
  metadata_failed: number;
  thumbnails_generated: number;
  thumbnail_failed: number;
};

export type VisualAnalysisProcessResponse = {
  job_id: number;
  total_photos: number;
  analyzed_photos: number;
  failed_photos: number;
  skipped_non_images: number;
};

export type SimilarityDetectionResponse = {
  job_id: number;
  total_media: number;
  exact_groups: number;
  similar_groups: number;
  grouped_items: number;
  skipped_without_hash: number;
};

export type CurationProcessResponse = {
  job_id: number;
  total_media: number;
  selected: number;
  alternative: number;
  rejected: number;
  manual_review: number;
  preserved_manual_overrides: number;
};

export type PhotoEnhancementRequest = {
  preset_slug: string;
};

export type PhotoEnhancementResponse = {
  job_id: number;
  total_selected: number;
  enhanced: number;
  skipped: number;
  failed: number;
  preset_slug: string;
};

export type MediaAnalysis = {
  sharpness_score: number | null;
  brightness_score: number | null;
  contrast_score: number | null;
  noise_score: number | null;
  exposure_score: number | null;
  overall_quality_score: number | null;
  perceptual_hash: string | null;
  analysis_version: string;
  raw_metrics_json: string | null;
  analyzed_at: string;
};

export type OriginalMedia = {
  id: number;
  event_id: number;
  source_id: number | null;
  original_path: string;
  relative_path: string | null;
  filename: string;
  extension: string | null;
  media_type: string;
  mime_type: string | null;
  file_size_bytes: number | null;
  checksum_sha256: string | null;
  capture_datetime: string | null;
  date_source: string | null;
  width: number | null;
  height: number | null;
  duration_seconds: number | null;
  thumbnail_path: string | null;
  thumbnail_url: string | null;
  metadata_json: string | null;
  analysis: MediaAnalysis | null;
  status: string;
  original_exists: boolean;
  imported_at: string;
};

export type OriginalMediaListResponse = {
  items: OriginalMedia[];
};

export type SimilarityGroupItem = {
  id: number;
  group_id: number;
  original_media_id: number;
  distance_score: number | null;
  role: string;
  reason: string | null;
  media: OriginalMedia;
};

export type SimilarityGroup = {
  id: number;
  event_id: number;
  group_type: string;
  representative_media_id: number | null;
  confidence_score: number | null;
  reason: string | null;
  created_at: string;
  items: SimilarityGroupItem[];
};

export type SimilarityGroupListResponse = {
  items: SimilarityGroup[];
};

export type CuratedMedia = {
  id: number;
  event_id: number;
  original_media_id: number;
  selection_status: string;
  reason: string | null;
  score: number | null;
  selected_by: string;
  is_manual_override: boolean;
  created_at: string;
  updated_at: string;
  media: OriginalMedia;
};

export type CuratedMediaListResponse = {
  items: CuratedMedia[];
};

export type CuratedMediaUpdate = {
  selection_status: string;
  reason?: string | null;
};

export type EnhancedMedia = {
  id: number;
  event_id: number;
  original_media_id: number;
  curated_media_id: number | null;
  output_path: string;
  output_url: string;
  enhancement_type: string;
  preset_slug: string | null;
  status: string;
  width: number | null;
  height: number | null;
  duration_seconds: number | null;
  notes: string | null;
  created_at: string;
  approved_at: string | null;
  rejected_at: string | null;
  media: OriginalMedia;
};

export type EnhancedMediaListResponse = {
  items: EnhancedMedia[];
};

export type EnhancedMediaUpdate = {
  status: string;
  reason?: string | null;
};

File: backend/app/schemas/importing.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class SourceCreate(BaseModel):
    source_path: str = Field(min_length=1, max_length=1024)


class SourceResponse(BaseModel):
    id: int
    event_id: int
    source_path: str
    source_type: str
    status: str
    file_count: int
    scanned_at: datetime | None
    imported_at: datetime | None
    notes: str | None
    created_at: datetime
    updated_at: datetime


class ScanSourceSummary(BaseModel):
    source_id: int
    source_path: str
    supported_files: int
    unsupported_files: int
    failed_files: int


class ScanResponse(BaseModel):
    job_id: int
    total_sources: int
    supported_files: int
    unsupported_files: int
    failed_files: int
    sources: list[ScanSourceSummary]


class ImportResponse(BaseModel):
    job_id: int
    imported_files: int
    skipped_files: int
    failed_files: int
    total_files: int


class MetadataProcessResponse(BaseModel):
    metadata_job_id: int
    thumbnail_job_id: int
    total_files: int
    metadata_updated: int
    metadata_failed: int
    thumbnails_generated: int
    thumbnail_failed: int


class VisualAnalysisProcessResponse(BaseModel):
    job_id: int
    total_photos: int
    analyzed_photos: int
    failed_photos: int
    skipped_non_images: int


class SimilarityDetectionResponse(BaseModel):
    job_id: int
    total_media: int
    exact_groups: int
    similar_groups: int
    grouped_items: int
    skipped_without_hash: int


class CurationProcessResponse(BaseModel):
    job_id: int
    total_media: int
    selected: int
    alternative: int
    rejected: int
    manual_review: int
    preserved_manual_overrides: int


class CuratedMediaUpdate(BaseModel):
    selection_status: str = Field(min_length=1, max_length=40)
    reason: str | None = Field(default=None, max_length=1000)


class MediaAnalysisResponse(BaseModel):
    sharpness_score: float | None
    brightness_score: float | None
    contrast_score: float | None
    noise_score: float | None
    exposure_score: float | None
    overall_quality_score: float | None
    perceptual_hash: str | None
    analysis_version: str
    raw_metrics_json: str | None
    analyzed_at: datetime


class OriginalMediaResponse(BaseModel):
    id: int
    event_id: int
    source_id: int | None
    original_path: str
    relative_path: str | None
    filename: str
    extension: str | None
    media_type: str
    mime_type: str | None
    file_size_bytes: int | None
    checksum_sha256: str | None
    capture_datetime: datetime | None
    date_source: str | None
    width: int | None
    height: int | None
    duration_seconds: float | None
    thumbnail_path: str | None
    thumbnail_url: str | None
    metadata_json: str | None
    analysis: MediaAnalysisResponse | None
    status: str
    original_exists: bool
    imported_at: datetime


class OriginalMediaListResponse(BaseModel):
    items: list[OriginalMediaResponse]


class SimilarityGroupItemResponse(BaseModel):
    id: int
    group_id: int
    original_media_id: int
    distance_score: float | None
    role: str
    reason: str | None
    media: OriginalMediaResponse


class SimilarityGroupResponse(BaseModel):
    id: int
    event_id: int
    group_type: str
    representative_media_id: int | None
    confidence_score: float | None
    reason: str | None
    created_at: datetime
    items: list[SimilarityGroupItemResponse]


class SimilarityGroupListResponse(BaseModel):
    items: list[SimilarityGroupResponse]


class CuratedMediaResponse(BaseModel):
    id: int
    event_id: int
    original_media_id: int
    selection_status: str
    reason: str | None
    score: float | None
    selected_by: str
    is_manual_override: bool
    created_at: datetime
    updated_at: datetime
    media: OriginalMediaResponse


class CuratedMediaListResponse(BaseModel):
    items: list[CuratedMediaResponse]
